collect_app/src/main/java/org/odk/collect/android/widgets/utilities/ExternalAppAudioDataRequester.java

```diff
@@ -0,0 +1,68 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.provider.MediaStore;
+import android.widget.Toast;
+
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.R;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.utilities.ActivityAvailability;
+import org.odk.collect.android.utilities.ApplicationConstants;
+import org.odk.collect.android.utilities.PermissionUtils;
+
+public class ExternalAppAudioDataRequester implements AudioDataRequester {
```

Comment:
We should be able to swap this implementation with another one for internal recording